jasper_server/obj_server.py

```diff
@@ -249,20 +250,31 @@ def generatorYAML(self, cr, uid, jasper_document, current_object, user_company,
         for yaml_object in jasper_document.yaml_object_ids:
 
             model_obj = self.pool.get(yaml_object.model.model)
-            model_ids = model_obj.search(cr, uid,
+            user_id = uid
+            if yaml_object.yaml_user_id.id:
+                user_id = yaml_object.yaml_user_id.id
+            if yaml_object.yaml_context:
+                yaml_context = ast.literal_eval( yaml_object.yaml_context)
+                for key,value in yaml_context.items():
+                    context[key] = value
```

Comment:
Please copy the context always before using it. Otherwise you change the original context